sky/shell/platform/ios/FlutterViewController.mm

```diff
@@ -183,17 +187,27 @@ - (void)setupPlatformServiceProvider {
   mojo::ServiceProviderPtr viewServiceProvider;
   new sky::shell::ViewServiceProvider(mojo::GetProxy(&viewServiceProvider));
 
+  mojo::ServiceProviderPtr outgoingServiceProvider;
+
   sky::ServicesDataPtr services = sky::ServicesData::New();
   services->incoming_services = serviceProvider.Pass();
+  services->outgoing_services = mojo::GetProxy(&outgoingServiceProvider);
   services->view_services = viewServiceProvider.Pass();
   _engine->SetServices(services.Pass());
+
+  semantics::SemanticsServerPtr semantics_server;
+  mojo::ConnectToService(outgoingServiceProvider.get(), &semantics_server);
+  _semanticsServer = semantics_server.Pass();
 }
 
 #pragma mark - Loading the view
 
 - (void)loadView {
   FlutterView* surface = [[FlutterView alloc] init];
 
+  // Construct the listener implementation
+  new sky::shell::SemanticsListenerImpl(surface, _semanticsServer.Pass());
```

Comment:
The controller may not always be a singleton as we are exposing the `FlutterViewController` in the upside-down world (where third parties can embed a Flutter view in their own hierarchy).

Granted, there are other issues around having multiple instances of the flutter engine active at the same time. But we will need to fix those in due course as well.